shared/test/test_varnish_helpers.rb

```diff
@@ -0,0 +1,130 @@
+require 'minitest/autorun'
+require '../../cookbooks/cdo-varnish/libraries/helpers'
+
+# Unit tests for cdo-varnish/libraries/helpers.rb
+class VarnishHelperTest < Minitest::Test
+
+  def test_paths_to_regex
+    ext_regex = paths_to_regex('*.jpg')
+    assert_equal 'req.url ~ "(\.jpg)(\?.*)?$"', ext_regex
+
+    assert_equal ext_regex, paths_to_regex(['*.jpg'])
+    # extension-wildcard with leading slash allowed
+    assert_equal ext_regex, paths_to_regex('/*.jpg')
+
+    exts_regex = paths_to_regex(%w(*.jpg *.png))
+    assert_equal 'req.url ~ "(\.jpg|\.png)(\?.*)?$"', exts_regex
+
+    path_regex = paths_to_regex(%w(path1/* path2/*))
+    assert_equal 'req.url ~ "^/path1/" || req.url ~ "^/path2/"',  path_regex
+  end
+
+  def test_invalid_path_patterns
+    [
+      # Unsupported extension wildcards
+      '/images/*.jpg',
+      '/a??.jpg',
+      '/a*.jpg',
+      # Wildcard-extension
+      '/*.doc*',
+      # Path with wildcard subexpression
+      '/assets/*/images',
+      # Extension-wildcard containing path delimiter
+      '*/images',
+      # Wildcard at both ends
+      '*images*',
+      # Maximum path length
```

Comment:
Could you add checks for characters like [] and ()?  These currently aren't being escape in regular expressions so it's important they don't occur in paths.